crates/sui-json/src/lib.rs

```diff
@@ -149,7 +156,7 @@ fn try_from_bcs_bytes(bytes: &[u8]) -> Result<JsonValue, anyhow::Error> {
     if let Ok(v) = bcs::from_bytes::<String>(bytes) {
         Ok(JsonValue::String(v))
     } else if let Ok(v) = bcs::from_bytes::<AccountAddress>(bytes) {
-        Ok(JsonValue::String(v.to_hex_literal()))
```

Comment:
This is a bug--if the account address has leading 0's, it will produce a JSON address that is too short (`to_hex_literal` call `AccountAddress::to_short_str_lossless()` under the hood) and that the `move_call` API will fail to process process.